src/ray/common/grpc_util.h

```diff
@@ -80,21 +80,19 @@ inline grpc::Status RayStatusToGrpcStatus(const Status &ray_status) {
   if (ray_status.ok()) {
     return grpc::Status::OK;
   } else {
-    // TODO(hchen): Use more specific error code.
-    return grpc::Status(grpc::StatusCode::UNKNOWN, ray_status.message());
+    return grpc::Status(grpc::StatusCode::UNKNOWN, ray_status.CodeAsString(),
```

Comment:
yeah the reason why I set ray status_code as grpc error_meesage is that there's already a `Serialize` method (which is `CodeAsString`) in codes.
Using a special grpc error code (either UNKNOWN or ABORTED is ok) to represent non-ok status and 
- status code as grpc error message
- status message as grpc error details
seems a better layering.
Thanks for opening discuss. Let's keep the current logic if you don't have strong opinion :)

Reply:
cc @scv119 @rkooo567 How much compatibility breakage is allowed here in future? This PR does not break compatibility, but it will make future changes to `ray::Status` likely breaking, e.g. if we want to have minor error codes, attach additional payloads, or a stack trace. If we are ok with breaking compatibility, I'm ok with this. Otherwise I prefer setting the `details` field in an extensible way, or not setting it at all.

How about setting grpc status message as `absl::StrCat(ray_status.CodeAsString(), ": ", ray_status.message())`? Then we don't need to set the details anymore.

Reply:
I can't see the difference between putting ray_status.message as part of grpc error message or as grpc error details.
It seems the compatibility breakage is same with either solution. Correct me if i am wrong.

Reply:
echo @mwtian's point the conversion is lossy right now. we lost the ray_status.message() which we might want add back later. 

Following suggestion makes sense to me.

> How about setting grpc status message as absl::StrCat(ray_status.CodeAsString(), ": ", ray_status.message())? Then we don't need to set the details anymore.

Reply:
```
we lost the ray_status.message() which we might want add back later.
```
We DON'T lost `ray_status.message()`.  It's set as grpc error_details in https://github.com/ray-project/ray/pull/20047/files#diff-3682c96d278137cbc42f115fcaf4e697ba20a4c9dee3fcd3df2b0406c8c4c914R84. Maybe you miss it.

Reply:
oh I missed that part. Then I think this looks good to me.

Reply:
I don't think we will care the backward compatibility of arbitrary 2 core components. Please let me know if we have some use cases here (I think this can be only useful if we plan to make different version of Ray communicate to each other)? @mwtian about the future compatibility to Ray::Status, do you have any change in your mind that can break the compatibility?

Reply:
<img width="1596" alt="Screen Shot 2021-11-05 at 4 24 52 AM" src="https://user-images.githubusercontent.com/18510752/140503460-92c579af-62a4-4e5b-80f6-ac75709b03c3.png">

Regarding the status code, according to this document, it seems like `Unknown` could be generated by gRPC library whereas `ABORTED` is not, so maybe ABORTED is a better option here?

Reply:
What I'm thinking is that if compatibility is needed and additional data need to be attached, it will be easier if `details` is not used or is a proto message. Compatibility can make features like rolling updates possible. It seems compatibility is not a concern per @rkooo567, then I'm ok with this.

Another thing to watch out is if any status printing / monitoring change will break because of this.